File: frontend/src/types/core/interfaces/property.ts
/**
 * Represents a property used to store details to describe the relationship shared between {@link Vertex} by an {@link Edge}. Stored inside of a {@link Graphset}.
 */
export interface Property {
	/**
	 * Unique identifier for the property.
	 */
	id: string;
	/**
   Label of name of the property.
   */
	label: string;
	/**
   Short description providing additional details about the property.
   */
	description: string;
}

File: backend/src/main/java/edu/velv/wikiverse_api/models/core/Property.java
package edu.velv.wikiverse_api.models.core;

public class Property {
  private String id;
  private String label;
  private String description;

  /**
   * Unique identifier for the property.
   */
  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  /**
   * Label or name of the property.
   */
  public String getLabel() {
    return label;
  }

  public void setLabel(String label) {
    this.label = label;
  }

  /**
   * Short description providing additional details about the property.
   */
  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }
}
